### ISSUE TITLE
Move Text.Megaparsec.Combinator to a separate package

### ISSUE THREAD

**Issue by mrkkrp** [mrkkrp/megaparsec#211]
The functions there are general enough to be in a separate package. That package should have no dependencies but `base`, be well-tested and benchmarked. We probably want to re-export `many`, `some` and `optional` from there as well, with additional documentation tailored for parsing.

When the package is released, we can depend on that in Megaparsec. Basically it's one more way to "unload" the main package and make it easier to maintain. I don't think that `parser-combinators` library will change often.

Basically it's in the same spirit as Kmett's `parsers`, but it should provide only truly general stuff and should not depend on `attoparsec` and `parsec`. This way anyone who is building a parser combinator library could benefit from those combinators without depending on other parsing libraries.

**Comment by mrkkrp** [comment]
`Text.Megaparec.Prim` should be moved to `Text.Megaparsec` and that module should re-export the combinators as well.
